test/python/transpiler/test_optimize_1q_decomposition.py

```diff
@@ -730,6 +732,52 @@ def test_prefer_no_substitution_if_all_ideal(self):
         res = opt_pass(qc)
         self.assertEqual(res, qc)
 
+    def test_custom_gate(self):
+        """Test that pass handles custom single-qubit gates."""
+
+        class CustomGate(Gate):
+            """Custom u1 gate."""
+
+            def __init__(self, lam):
+                super().__init__("custom_u1", 1, [lam])
+
+            def __array__(self, dtype=None, _copy=None):
+                return U1Gate(*self.params).__array__(dtype=dtype)
+
+        qc = QuantumCircuit(1)
+        qc.append(CustomGate(0.5), [0])
+        result = Optimize1qGatesDecomposition(["cx", "u"])(qc)
+
+        expected = QuantumCircuit(1)
+        expected.u(0, 0, 0.5, [0])
+
+        self.assertEqual(result, expected)
+
+    def test_unitary_gate_ndarray(self):
+        """Test that pass handles unitary single-qubit gates."""
+
+        qc = QuantumCircuit(1)
+        # The Rust code is not able to interpret mat as nalgebra matrix
+        mat = np.asarray(SGate().to_matrix(), dtype=complex)
```

Comment:
No need to change but I think you can also just do
```suggestion
        mat = np.array(SGate())
```
since `SGate` implements `__array__`

Reply:
Great to know, thanks (when you give me a choice, I will take the "no need to change option", haha).